Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -401,45 +358,30 @@ open class MapView: UIView {
         }
     }
 
-    private func updateDisplayLinkPreferredFramesPerSecond() {
-        if let displayLink = displayLink {
-            if let _preferredFramesPerSecond = _preferredFramesPerSecond {
-                displayLink.preferredFramesPerSecond = _preferredFramesPerSecond
-            }
-            // Checking Swift version as a proxy for iOS SDK version to enable
-            // building with iOS SDKs < 15
-            #if swift(>=5.5)
-            if #available(iOS 15.0, *) {
-                if let _preferredFrameRateRange = _preferredFrameRateRange {
-                    displayLink.preferredFrameRateRange = _preferredFrameRateRange
-                }
-            }
-            #endif
-        }
-    }
-
-    open override func willMove(toWindow newWindow: UIWindow?) {
-        super.willMove(toWindow: newWindow)
-        if newWindow != nil {
-            validateDisplayLink()
-        }
-    }
-
     open override func didMoveToWindow() {
         super.didMoveToWindow()
 
-        if window != nil {
-            validateDisplayLink()
-        } else {
-            // TODO: Fix this up correctly.
-            displayLink?.invalidate()
-            displayLink = nil
+        displayLink.invalidate()
+
+        let oldDisplayLink = displayLink
+        let newDisplayLink = dependencyProvider.makeDisplayLink(
+            window: window,
+            target: ForwardingDisplayLinkTarget { [weak self] in
+                self?.updateFromDisplayLink(displayLink: $0)
+            },
+            selector: #selector(ForwardingDisplayLinkTarget.update(with:)))
+        displayLink = newDisplayLink ?? EmptyDisplayLink()
+
+        // Checking Swift version as a proxy for iOS SDK version to enable
+        // building with iOS SDKs < 15
+#if swift(>=5.5)
+        if #available(iOS 15.0, *) {
+            displayLink.preferredFrameRateRange = oldDisplayLink.preferredFrameRateRange
         }
-    }
+#endif
+        displayLink.preferredFramesPerSecond = oldDisplayLink.preferredFramesPerSecond
```

Comment:
What happens if these properties on MapView were never set? Will this overwrite the default values for the screen?

Reply:
Yes indeed, when map view did move to window happens frame rate values for a new 'CADisplayLink' will be overridden with corresponding values from 'EmptyDisplayLink'.

Reply:
The display link is created via the screen as that's the only way to have the link linked to the correct screen in multiscreen setup. The default values in `EmptyDisplayLink` are the same as ones from newly created `CADisplayLink`, so overriding should not affect anything.